src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -288,6 +296,30 @@ async def _add_objects_to_archive(
     return pack_result
 
 
+def _maybe_is_golang_assembly(data: bytes) -> bool:
+    return (
+        data.startswith(b"TEXT")
+        or b"\nTEXT" in data
+        or data.startswith(b"DATA")
+        or b"\nDATA" in data
+        or data.startswith(b"GLOBL")
+        or b"\nGLOBL" in data
+    )
+
+
+@rule_helper
+async def _any_file_is_golang_assembly(
+    digest: Digest, dir_path: str, s_files: Iterable[str]
+) -> bool:
+    digest_contents = await Get(DigestContents, Digest, digest)
```

Comment:
Will the `s_files` be a small enough proportion of the files for it to be worth doing a `DigestSubset` before loading all content into memory?

Reply:
Updated this to use `DigestSubset`.